File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <algorithm>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> temp_arr = arr;
    vector<int> answer;
    
    sort(temp_arr.rbegin(), temp_arr.rend());
    int small = temp_arr[temp_arr.size()-1];

    for(int i=0; i<arr.size(); i++) if (arr[i] != small) answer.push_back(arr[i]);
    if (answer.size() == 0) answer.push_back(-1);
    
    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(arr) {
    
    let tmp_arr = [...arr];
    tmp_arr = tmp_arr.sort((a,b) => a-b);
    const valueToRemove = tmp_arr[0];
    const newArr = arr.filter((item)=> item !== valueToRemove);
    
    if (newArr.length == 0) return [-1];
    else return newArr;
}